### PR TITLE
Feature: hard delete message event

### PR DESCRIPTION
**Pull request by mahboubii**

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -1496,9 +1497,14 @@ export class Channel<
           s.watchers = s.watchers.without(event.user.id);
         }
         break;
+      case 'message.deleted':
+        if (event.message) {
+          if (event.hard_delete) s.removeMessage(event.message as { id: string });
```
Comment (DanC5) [inline src/channel.ts:1502]: casting because id in MessageBase is possibly undefined?

File: src/channel.ts [hunk 2]
```diff
@@ -1496,9 +1497,14 @@ export class Channel<
...
+          else s.addMessageSorted(event.message);
```
Comment (DanC5) [inline src/channel.ts:1503]: i wonder if we should update the `removeMessage` type parameter to match `addMessageSorted` so we don't have to cast, unless there's a reason it should stay as it is

File: src/channel.ts (hunk 1 above)
Comment (mahboubii) [reply src/channel.ts:1502]: yea, it's either checking the id for undefined(which somehow TS was failing to understand) or casting the message type

Comment (DanC5) [reply src/channel.ts:1502]: i guess you could do `if (event.hard_delete && event.message.id)` if we don't want to change the type

File: src/channel.ts (hunk 2 above)
Comment (mahboubii) [reply src/channel.ts:1503]: actually, `removeMessage` param type is more correct than `addMessageSorted`, we should make the  `id` required in `MessageBase` type

Comment (DanC5) [reply src/channel.ts:1503]: it definitely was required at one point, not sure why it turned optional, @nhannah do you remember?

Comment (mahboubii) [reply src/channel.ts:1503]: swapped it to required

File: src/channel.ts (hunk 1 above)
Comment (mahboubii) [reply src/channel.ts]: weirdly yea

File: src/channel.ts (hunk 2 above)
Comment (DanC5) [reply src/channel.ts:1503]: great! :vir_stache:, let's pretend that worked...

**Review by DanC5 — APPROVED** [review]

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**